src/coreclr/jit/morph.cpp

```diff
@@ -11161,7 +11181,7 @@ GenTree* Compiler::fgMorphSmpOp(GenTree* tree, MorphAddrContext* mac, bool* optA
             //
             if (!tree->AsIndir()->IsVolatile())
             {
-                if (op1->IsIconHandle(GTF_ICON_STR_HDL))
+                if (op1->IsIconHandle(GTF_ICON_OBJ_HDL))
```

Comment:
I guess we see this for cases like `"" is string`?

Reply:
I bet this can be removed since we now add these flags in morph anyway in `case GT_IND:` and the `is volatile` for string literal load seems to be useless - I'll check it in a follow up together with https://github.com/dotnet/runtime/pull/75573/files#r977367516